x/upgrade/abci_test.go

```diff
@@ -4,182 +4,394 @@ import (
+func VerifyCleared(t *testing.T, newCtx sdk.Context) {
+	t.Log("Verify that the upgrade plan has been cleared")
 	bz, err := s.querier(newCtx, []string{upgrade.QueryCurrent}, abci.RequestQuery{})
-	s.Require().NoError(err)
-	s.Require().Nil(bz)
+	require.NoError(t, err)
+	require.Nil(t, bz)
 }
 
-func (s *TestSuite) TestCanClear() {
-	s.T().Log("Verify upgrade is scheduled")
+func TestCanClear(t *testing.T) {
+	s := setupTest(10, map[int64]bool{})
+	t.Log("Verify upgrade is scheduled")
 	err := s.handler(s.ctx, upgrade.SoftwareUpgradeProposal{Title: "prop", Plan: upgrade.Plan{Name: "test", Time: time.Now()}})
-	s.Require().Nil(err)
+	require.Nil(t, err)
 
-	s.handler(s.ctx, upgrade.CancelSoftwareUpgradeProposal{Title: "cancel"})
+	err = s.handler(s.ctx, upgrade.CancelSoftwareUpgradeProposal{Title: "cancel"})
+	require.Nil(t, err)
 
-	s.VerifyCleared(s.ctx)
+	VerifyCleared(t, s.ctx)
 }
 
-func (s *TestSuite) TestCantApplySameUpgradeTwice() {
-	s.TestDoTimeUpgrade()
-	s.T().Log("Verify an upgrade named \"test\" can't be scheduled twice")
+func TestCantApplySameUpgradeTwice(t *testing.T) {
+	s := setupTest(10, map[int64]bool{})
 	err := s.handler(s.ctx, upgrade.SoftwareUpgradeProposal{Title: "prop", Plan: upgrade.Plan{Name: "test", Time: time.Now()}})
-	s.Require().NotNil(err)
-	s.Require().Equal(sdk.CodeUnknownRequest, err.Code())
+	require.Nil(t, err)
+	VerifyDoUpgrade(t)
+	t.Log("Verify an executed upgrade \"test\" can't be rescheduled")
+	err = s.handler(s.ctx, upgrade.SoftwareUpgradeProposal{Title: "prop", Plan: upgrade.Plan{Name: "test", Time: time.Now()}})
+	require.NotNil(t, err)
+	require.Equal(t, sdk.CodeUnknownRequest, err.Code())
 }
 
-func (s *TestSuite) TestNoSpuriousUpgrades() {
-	s.T().Log("Verify that no upgrade panic is triggered in the BeginBlocker when we haven't scheduled an upgrade")
+func TestNoSpuriousUpgrades(t *testing.T) {
+	s := setupTest(10, map[int64]bool{})
+	t.Log("Verify that no upgrade panic is triggered in the BeginBlocker when we haven't scheduled an upgrade")
 	req := abci.RequestBeginBlock{Header: s.ctx.BlockHeader()}
-	s.Require().NotPanics(func() {
+	require.NotPanics(t, func() {
 		s.module.BeginBlock(s.ctx, req)
 	})
 }
 
-func (s *TestSuite) TestPlanStringer() {
-	t, err := time.Parse(time.RFC3339, "2020-01-01T00:00:00Z")
-	s.Require().Nil(err)
-	s.Require().Equal(`Upgrade Plan
+func TestPlanStringer(t *testing.T) {
+	ti, err := time.Parse(time.RFC3339, "2020-01-01T00:00:00Z")
+	require.Nil(t, err)
+	require.Equal(t, `Upgrade Plan
   Name: test
   Time: 2020-01-01T00:00:00Z
-  Info: `, upgrade.Plan{Name: "test", Time: t}.String())
-	s.Require().Equal(`Upgrade Plan
+  Info: `, upgrade.Plan{Name: "test", Time: ti}.String())
+	require.Equal(t, `Upgrade Plan
   Name: test
   Height: 100
   Info: `, upgrade.Plan{Name: "test", Height: 100}.String())
 }
 
-func TestTestSuite(t *testing.T) {
-	suite.Run(t, new(TestSuite))
+func VerifyNotDone(t *testing.T, newCtx sdk.Context, name string) {
```

Comment:
nit/non-blocking: the structure & style of tests and comments can definitely be cleaned up. e.g. 

- space between `//` and start comment
- avoid punctuation unless it's a sentence 
- avoid group var statements when you assign values
- remove commented-out code

Reply:
Noted. We'll ensure these things are taken care from next time. Any specific reason to `avoid group var statements`?

Reply:
```go
var (
  foo int64 = 4
  bar bool = true
)
```

Instead,

```go
foo := 4 // or foo := int64(4)
bar := true
```

Reply:
I understand. Just wanted to know if there's any specific reason to avoid group declarations. I think having group declartion for this particular scenario is needed as it is making more readable. It is creating a bit of inconsistency in the code, too. We can take a call.

Reply:
I understand. Just wanted to know if there's any specific reason to avoid group declarations. I think having group declartion for this particular scenario is needed as it is making more readable. It is creating a bit of inconsistency in the code, too. We can take a call.